examples/browser/rocket_trajectory_optimization.html

```diff
@@ -11,142 +12,290 @@
-          pointRadius: 0
-        }, {
-          label: "Unpowered",
-          data: data_unpowered,
-          fill: false,
-          borderColor: "blue",
-          pointRadius: 0
+      data: { datasets: mainDatasets },
+      options: getMainChartOptions()
+    })
+    createChart([{
+      label: 'velocity (in m/s)',
+      data: sim.evaluate("result[:,[2,6]]")
+        .toArray()
+        .map(([v, t]) => ({ x: t, y: v.toNumber('m/s') }))
+    }])
+    createChart([{
+      label: 'height (in km)',
+      data: sim.evaluate("concat((result[:, 1] - r0)/km, result[:, 6])")
+        .toArray()
+        .map(([r, t]) => ({ x: t, y: r })),
+    }])
+    createChart([{
+      label: 'gamma (in deg)',
+      data: sim.evaluate("result[:, [5,6]]")
+        .toArray()
+        .map(([gamma, t]) => ({ x: t, y: gamma.toNumber('deg') })),
+    }])
+    createChart([{
+      label: 'acceleration (in m/s^2)',
+      data: sim.evaluate("concat(diff(result[:, 2]) ./ diff(result[:, 6]) / (m/s), result[:end-1, 6])")
+        .toArray()
+        .map(([acc, t]) => ({ x: t, y: acc })),
+    }])
+    createChart([{
+      label: 'drag acceleration (in m/s^2)',
+      data: sim.evaluate("concat(drag(result[:, 1], result[:, 2]) ./ result[:, 3] / (m/s^2), result[:, 6])")
+        .toArray()
+        .map(([dragAcc, t]) => ({ x: t, y: dragAcc })),
+    }])
+    createChart([{
+      data: sim.evaluate("result[:, [1,4]]")
+        .toArray()
+        .map(([r, phi]) => ([r.toNumber('km'), phi.toNumber('rad')]))
+        .map(([r, phi]) => ({ x: r * Math.cos(phi), y: r * Math.sin(phi) })),
+    }, {
+      data: sim.evaluate("map(0:0.25:360, function(angle) = rotate([r0/km, 0], angle))")
+        .toArray()
+        .map(([x, y]) => ({ x, y })),
+      borderColor: "green",
+      fill: true
+    }], {
+      aspectRatio: 1,
+      scales: {
+        xAxes: [{
+          type: 'linear',
+          position: 'bottom',
+          min: -8000,
+          max: 8000,
+          display: false
+        }],
+        yAxes: [{
+          type: 'linear',
+          min: -8000,
+          max: 8000,
+          display: false
         }]
       },
-      options: {
+      legend: { display: false }
+    })
+
+    // This function is a hack, but I like the charts I can draw now!
+    function combineResults() {
+      let time = 0
+      let result
+      for (const stageName of stageNames) {
+        const resultName = 'result_' + stageName
+        const dt = resultName.includes('unpowered') ? '10' : '0.5' // This is a hack!
+        const resultMatrix = sim.evaluate(resultName)
+        const resultLength = resultMatrix.size()[0]
+        const startTime = time
+        time += dt * resultLength
+        const timeRange = math.range(startTime, time, dt)
+        const resultWithTime = math.concat(resultMatrix, math.reshape(timeRange, [resultLength, 1]))
+        result = result ? math.concat(result, resultWithTime, 0) : resultWithTime
+      }
+      return result
+    }
```

Comment:
Here time is added to the results matrix.
The results don't know about time and there are different `dt`s for different stages. 
My best idea is to make it cleaner is this:
* To record the time in the solution, add another equation to the solver: `dtdt(...) = 1`
* Rename `tfinal` to something more precise (like `tDuration`, `simulationTime`, `tStage`, ...)

And maybe `concat` should be called only once at the end.